moj-docker-deploy/apps/proxied-containers.sls

```diff
@@ -26,6 +26,21 @@ include:
     - watch_in:
       - service: nginx
 
+{% if appdata.get('basic_auth', False) %}
+/etc/nginx/conf.d/{{server_name}}.htpasswd
+  file.managed:
+    - source: salt//moj-docker-deploy/apps/templates/nginx_container.htpasswd
+    - user: root
+    - group: root
+    - mode: 644
+    - template: jinja
+    - context:
+      server_name: {{server_name}}
+      appdata: {{appdata | yaml}}
+    - watch_in:
+      - service: nginx
+{% endif %}
```

Comment:
You should have an else here with file.absent to remove this file if basic auth is removed